Workflow API docs (#2915)
dapr/docs: daprdocs/content/en/concepts/components-concept.md

- [List of supported locks]({{< ref supported-locks >}})
- [Lock implementations](https://github.com/dapr/components-contrib/tree/master/lock)

### Workflows

> A [workflow]({{< ref workflow-overview.md >}}) is custom application logic that defines business process or data flow in a reliable way across mulitple microservices. The workflow API is exposed by a [lightweight, embedded workflow engine]({{< ref "operations/components/workflow-engine/workflow-engine.md" >}}) in the Dapr sidecar, allowing you to easily integrate with existing Dapr building blocks. 

Reviewer: > The workflow API is exposed by a [lightweight, embedded workflow engine]({{< ref "operations/components/workflow-engine/workflow-engine.md" >}}) in the Dapr sidecar, allowing you to easily integrate with existing Dapr building blocks

@RyanLettieri Could you add more context here? Is the workflow API exposed through the workflow engine - will users need the workflow engine in place to use the workflow API (even if with Temporal)?

> A [workflow]({{< ref workflow-overview.md >}}) is custom application logic that defines business process or data flow in a reliable way across mulitple microservices. The workflow API is exposed by a [lightweight, embedded workflow engine]({{< ref "operations/components/workflow-engine/workflow-engine.md" >}}) in the Dapr sidecar, allowing you to easily integrate with existing Dapr building blocks. 

Reviewer 2: This section is about workflow components, of which there is one and that is Temporal. 

```suggestion
A [workflow]({{< ref workflow-overview.md >}}) is custom application logic that defines a reliable business process or data flow. Workflow components are workflow runtimes (or engines) that run the business logic written for that workflow and store their state into a state store. 
```

> - [List of supported workflows]({{< ref supported-workflows >}})

Reviewer: ```suggestion
- [List of supported workflow components]({{< ref supported-workflows >}})
```

> - Workflow implementations

Reviewer 2: ```suggestion
- [Workflow implementations](- [Binding implementations](https://github.com/dapr/components-contrib/tree/master/workflows)
```
